src/storage/transaction/TransactionManager.cpp

```diff
@@ -8,414 +8,195 @@
 
 #include <folly/container/Enumerate.h>
 
-#include "clients/storage/InternalStorageClient.h"
 #include "codec/RowWriterV2.h"
 #include "common/utils/NebulaKeyUtils.h"
+#include "kvstore/NebulaStore.h"
 #include "storage/CommonUtils.h"
 #include "storage/StorageFlags.h"
-#include "storage/mutate/AddEdgesProcessor.h"
-#include "storage/transaction/TransactionUtils.h"
+#include "storage/transaction/ChainResumeProcessor.h"
 
 namespace nebula {
 namespace storage {
 
-/*
- * edgeKey : thrift data structure
- * rawKey  : NebulaKeyUtils::edgeKey
- * lockKey : rawKey + lock suffix
- * */
+DEFINE_int32(resume_interval_secs, 10, "Resume interval");
+
+ProcessorCounters kForwardTranxCounters;
+
 TransactionManager::TransactionManager(StorageEnv* env) : env_(env) {
   exec_ = std::make_shared<folly::IOThreadPoolExecutor>(10);
-  interClient_ = std::make_unique<storage::InternalStorageClient>(exec_, env_->metaClient_);
+  iClient_ = env_->interClient_;
+  resumeThread_ = std::make_unique<thread::GenericWorker>();
+  scanAll();
+  auto fn = std::bind(&TransactionManager::onNewPartAdded, this, std::placeholders::_1);
+  static_cast<::nebula::kvstore::NebulaStore*>(env_->kvstore_)
+      ->registerOnNewPartAdded("TransactionManager", fn);
 }
 
-/*
- * multi edges have same local partition and same remote partition
- * will process as a batch
- * *Important**Important**Important*
- * normally, ver will be 0
- * */
-folly::Future<nebula::cpp2::ErrorCode> TransactionManager::addSamePartEdges(
-    size_t vIdLen,
-    GraphSpaceID spaceId,
-    PartitionID localPart,
-    PartitionID remotePart,
-    std::vector<KV>& localEdges,
-    AddEdgesProcessor* processor,
-    folly::Optional<GetBatchFunc> optBatchGetter) {
-  int64_t txnId = TransactionUtils::getSnowFlakeUUID();
-  if (FLAGS_trace_toss) {
-    for (auto& kv : localEdges) {
-      auto srcId = NebulaKeyUtils::getSrcId(vIdLen, kv.first);
-      auto dstId = NebulaKeyUtils::getDstId(vIdLen, kv.first);
-      LOG(INFO) << "begin txn hexSrcDst=" << folly::hexlify(srcId) << folly::hexlify(dstId)
-                << ", txnId=" << txnId;
-    }
+TransactionManager::LockCore* TransactionManager::getLockCore(GraphSpaceID spaceId,
+                                                              GraphSpaceID partId) {
+  if (whiteListParts_.find(std::make_pair(spaceId, partId)) == whiteListParts_.end()) {
+    LOG(WARNING) << folly::sformat("space {}, part{} not in white list");
+    scanPrimes(spaceId, partId);
+    auto key = std::make_pair(spaceId, partId);
+    whiteListParts_.insert(std::make_pair(key, 0));
+    return nullptr;
```

Comment:
Pls check the logic around whiteListParts_

Reply:
done